pilot/pkg/model/context.go

```diff
@@ -302,6 +302,16 @@ type Proxy struct {
 	XdsNode *core.Node
 
 	AutoregisteredWorkloadEntryName string
+
+	// LastPushContext stores the most recent push context for this proxy. This will be monotonically
+	// increasing in version. Requests should send config based on this context; not the global latest.
+	// Historically, the latest was used which can cause problems when computing whether a push is
+	// required, as the computed sidecar scope version would not monotonically increase.
+	LastPushContext *PushContext
+	// LastPushTime records the time of the last push. This is used in conjunction with
+	// LastPushContext; the XDS cache depends on knowing the time of the PushContext to determine if a
+	// key is stale or not.
+	LastPushTime time.Time
```

Comment:
instead of two separate fields, I am thinking we should make time/token in PushContext. Originally I want to use push version as token, however it is not an ever-increasing integer.

Reply:
I was considering that.... one difference is this is per PushReuest vs per PushContext. I cannot think of any reason this would differ though so I can try that out